configure-windows-visual-studio.bat

```diff
@@ -261,8 +272,41 @@ set /p PTHREAD_HOME=    Please specify pthreads-win32 path: %=%
 :: if input dir doesn't exist print an error to the user and go back to previous line
 if not exist "%PTHREAD_HOME%"\ (echo Directory does not exist!! && goto while2)
 
+echo.
+echo.
+
+:: ask user about using zstd
+echo ZStd compression support may be added when compiling PcapPlusPlus.
+echo For downloading ZStd SDK (developer's pack) please go to https://github.com/facebook/zstd/releases
+:while3
+set /p USE_ZSTD=     To enable ZStd support type use_zstd otherwise type no: %=%
+if "%USE_ZSTD%" NEQ "use_zstd" if "%USE_ZSTD%" NEQ "no" (echo Please choose one of "use_zstd" or "no" && goto while3)
+
+
+if "%USE_ZSTD%" EQU "use_zstd" (
+::Can't use set staement inside an if so we have to do a little trickery here
+goto while4
+)
+goto skipWhile4
+
+:while4
+:: ask the user to type WinPcap/Npcap SDK dir
+set /p ZSTD_HOME=    Please specify ZStd SDK path: %=%
+:: if input dir doesn't exist print an error to the user and go back to previous line
+if not exist "%ZSTD_HOME%"\ (echo Directory does not exist!! && goto while4)
+set USE_ZSTD=USE_Z_STD;
+set "ZSTD_INCLUDE_PATH=$(ZStdHome)\include;"
+set ZSTD_LIB_NAME=libzstd.lib;
+
+:skipWhile4
+
+if "%USE_ZSTD%" EQU "no" set USE_ZSTD=
+
+echo.
+echo.
+
```

Comment:
we should add zstd to the command-line option too (lines 165-234)

Reply:
When I originally skipped this I didn't realize you were using it for your automated builds, so I agree its necessary. I will do it.

Reply:
done

Reply:
In second thought, maybe we should not add zstd to the step-by-step configuration because most user will probably not use it. I suggest adding it only as a command-line option. Please let me know what you think

Reply:
I think its presumptuous to assume most people won't use it.

1. I think this is (will be) a unique + powerful feature of this library and you should advertise it as such.
2. On average I am seeing about 4x space savings I have 140MB pcap files compressed down to 35MB! This is a HUGE savings and comes at nearly no overhead as it is done live during capture not post processing in bulk which is slow. I chose ZStd because it is well suited for streaming compression.
3. Integration is VERY easy they don't have to build ZStd. There are pre-built binaries just download them and point the parameter at the correct path.
4. The compression is over the whole file NOT per packet. Therefore you do NOT need to use pcap++ do decode it. Any ZStd compatible decompressor can dump out the original pcapng file. So users are not losing compatibility with any pcapng reader. 

Overall I think if you unveil this in an official release and advertise it well in your capabilities I think a lot of people will consider trying it and will be pleasantly surprised at the huge space savings.

Reply:
I agree with you that it's a very powerful feature that people might find useful. Thank you for the efforts you're putting into this and for your great contribution to this project.

However based on my experience, different users use this library for different purposes. Not all of them use pcapng, and in fact many of them don't read or write to files at all.

One thing that is very important to me as the library maintainer (and as a user of other open source libraries) is the simplicity and ease of use of the library. That includes installation, configuration, straight-forward design, and more. To put it in simple words, I'm looking to optimize the time it takes for an average user to have a simple code example up and running, assuming he/she starts from scratch, on any platform.

That's why I want to have the simplest installation and configuration process with minimum dependencies. Zstd is definitely a feature for more advanced users and that's why my suggestion would be to put it in the command-line configuration only.

 BTW, I've already done so for many advance features in Linux and MacOS, for example: `--use-immediate-mode`, `--set-direction-enabled`, `--install-dir`, `--libpcap-include-dir`, `--libpcap-lib-dir`.

Reply:
I totally understand your opinion and I agree this is a more advanced feature, and it is only applicable for people writing pcapng files.  However, IMO I don't think it overly onerous to put NO when prompted. I can also add notes in the prompt that indicate it will only benefit pcapng files in addition to the existing notes that it is optional.

Anyways at the end of the day the decision is in your hands as you are the chief maintainer/owner. I will make the change if requested, I was just sharing my opinion as originally asked.